hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java

```diff
@@ -54,192 +57,208 @@
+            // so later during the handle phase. Technically, we could still try to gather and verify the other
+            // signatures, but that might be tricky and complicated with little gain. So just throw.
+            return preHandleFailure(creator, PAYER_ACCOUNT_NOT_FOUND, txInfo, null);
+        }
 
-            // 3. Prepare and verify signature-data
-            final var signatureMap = transactionInfo.signatureMap();
-            final var txBodyBytes = transactionInfo.transaction().bodyBytes();
-            final var payerSignature = verifyPayerSignature(state, context, txBodyBytes, signatureMap);
-            final var otherSignatures = verifyOtherSignatures(state, context, txBodyBytes, signatureMap);
-
-            // 4. Eventually prepare and verify signatures of inner transaction
-            final var innerContext = context.innerContext();
-            PreHandleResult innerResult = null;
-            if (innerContext != null) {
-                // VERIFY: the txBytes used for inner transactions is the same as the outer transaction
-                final var innerPayerSignature = verifyPayerSignature(state, innerContext, txBytes, signatureMap);
-                final var innerOtherSignatures = verifyOtherSignatures(state, innerContext, txBytes, signatureMap);
-                innerResult = createResult(innerContext, signatureMap, innerPayerSignature, innerOtherSignatures, null);
-            }
+        // 3. Collect payer signature checks. Any PreHandleResult created from this point on MUST include
+        //    the payerVerificationFuture.
+        final var sigPairs = txInfo.signatureMap().sigPairOrThrow();
+        final var payerVerificationFuture = isHollow(payerAccount)
+                ? signatureVerifier.verify(payerAccount, txInfo.signedBytes(), sigPairs)
+                : signatureVerifier.verify(payerAccount.keyOrThrow(), txInfo.signedBytes(), sigPairs);
 
-            // 5. Return PreHandleResult
-            return createResult(context, signatureMap, payerSignature, otherSignatures, innerResult);
-
-        } catch (PreCheckException preCheckException) {
-            return createInvalidResult(preCheckException.responseCode());
-        } catch (Exception ex) {
-            // Some unknown and unexpected failure happened. If this was non-deterministic, I could
-            // end up with an ISS. It is critical that I log whatever happened, because we should
-            // have caught all legitimate failures in another catch block.
-            LOG.error("An unexpected exception was thrown during pre-handle", ex);
-            return createInvalidResult(UNKNOWN);
+        // 4. Create the PreHandleContext. This will get reused across several calls to the transaction handlers
+        final PreHandleContext context;
+        try {
+            // NOTE: Once PreHandleContext is moved from being a concrete implementation in SPI, to being an Interface/
```

Comment:
Well, `PreHandleContext` has already become an interface... 🙂

Reply:
Honestly I would fix this comment right now but if I do I lose all my approvals on the PR 😡 .